meteor/map_utils.py

```diff
@@ -0,0 +1,216 @@
+            + 2 * hl_prod * params[5]
+            + 2 * kl_prod * params[6]
+        )
+
+        return x_ref - params[0] * np.exp(t) * x_scale
+
+    reference_data = reference.to_numpy()
+    scale_data = dataset_to_scale.to_numpy()
+    uncertainties_data = uncertainties.to_numpy() if uncertainties is not None else None
+
+    miller_indices_ref = np.array(list(reference.index))
+    miller_indices_scale = np.array(list(dataset_to_scale.index))
+
+    assert np.array_equal(
+        miller_indices_ref, miller_indices_scale
+    ), "Miller indices of reference and dataset_to_scale do not match."  # noqa: E501
+
+    # Initial guess for: [C, B11, B22, B33, B12, B13, B23]
+    initial_params = np.array([1.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0], dtype=np.float32)
+
+    result = opt.least_squares(
+        aniso_scale_func,
+        initial_params,
+        args=(reference_data, scale_data, miller_indices_scale),
+    )
+
+    # Apply the scaling to dataset_to_scale
+    h, k, l = miller_indices_scale[:, 0], miller_indices_scale[:, 1], miller_indices_scale[:, 2]  # noqa: E741
+    h_sq, k_sq, l_sq = np.square(h), np.square(k), np.square(l)
+    hk_prod, hl_prod, kl_prod = h * k, h * l, k * l
+
+    t = -(
+        h_sq * result.x[1]
+        + k_sq * result.x[2]
+        + l_sq * result.x[3]
+        + 2 * hk_prod * result.x[4]
+        + 2 * hl_prod * result.x[5]
+        + 2 * kl_prod * result.x[6]
+    )
+
+    scale_factor = result.x[0] * np.exp(t)
+    scaled_data = scale_factor * scale_data
+
+    if uncertainties is not None:
+        scaled_uncertainties = scale_factor * uncertainties_data
+    else:
+        scaled_uncertainties = None
+
+    if inplace:
+        dataset_to_scale[:] = scaled_data
+        if uncertainties is not None:
+            uncertainties[:] = scaled_uncertainties
+        return None
+    else:
+        scaled_dataset = dataset_to_scale.copy()
+        scaled_dataset[:] = scaled_data
+        if uncertainties is not None:
+            scaled_uncertainties_series = uncertainties.copy()
+            scaled_uncertainties_series[:] = scaled_uncertainties
+            return scaled_dataset, scaled_uncertainties_series
+        return scaled_dataset
+
+def compute_amplitude_fofo_difference(
+    data1: rs.DataSeries,
+    data2: rs.DataSeries,
+    data3: rs.DataSeries,
+    uncertainties1: Optional[rs.DataSeries] = None,
+    uncertainties2: Optional[rs.DataSeries] = None,
+) -> Union[rs.DataSeries, tuple[rs.DataSeries, rs.DataSeries]]:
+    """
+    First, scale data1 and data2 to the common scale defined by data3,
+    then compute the difference (data2 - data1). Optionally, propagate uncertainties.
+
+    Parameters:
+    data1 (rs.DataSeries): First dataset to be used in difference calculation (e.g., F_off).
+    data2 (rs.DataSeries): Second dataset to be used in difference calculation (e.g., F_on).
+    data3 (rs.DataSeries): Reference dataset used for scaling data1 and data2 (e.g., F_calc).
```

Comment:
why can't we just scale two datasets to one another?? I don't think we need F_calc do we. Marius does this craziness but I never understood why.

Reply:
<img width="1185" alt="image" src="https://github.com/user-attachments/assets/55d9ce28-427e-4d38-be0e-4f9c651eb3a0">

I admit I inherited this from Marius, but intuitively, I assumed that  a common reference dataset would help in normalizing  relative to a stable baseline if the two datasets are noisy.

I ran a small test with two datasets at different noise levels

```
x = np.linspace(0, 10, 100)  # Common x-axis
y1 = 2 * x + 5 + np.random.normal(0, 1, 100)  
y2 = 2 * x + 5 + np.random.normal(0, 5, 100)
```

and scaled using 

```
def scale_to_each_other(y1, y2):
    coef = np.polyfit(y1, y2, 1)  # Fit y2 as a function of y1
    y2_scaled = coef[0] * y1 + coef[1]  # Scale y2 to y1
    return y2_scaled, coef
```
first to each other, then to the perfect `reference = 2 * x + 5 `. Seems like it is indeed better to scale to the common reference?